From 164f0e8d9945b943968886cfd90046f68bdda8be Mon Sep 17 00:00:00 2001
From: Abel Vesa <abel.vesa@nxp.com>
Date: Mon, 27 Nov 2017 18:10:11 +0200
Subject: [PATCH 2921/5242] MLK-17005: linux-imx: serial: fsl_lpuart: Clear
 LOOP mode when requested

commit  f64f6c1a61335809f3aa0e77d5424c34cb609863 from
https://source.codeaurora.org/external/imx/linux-imx.git

The LOOP mode remained always set after first use.
If the ioctl tiocmset gets called with TIOCMBIC for TIOCM_LOOP,
UARTCTRL_LOOPS needs to be cleared in the LPUART control register.

Signed-off-by: Abel Vesa <abel.vesa@nxp.com>
Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index 2663639..b37397d 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -1268,6 +1268,9 @@ static void lpuart32_set_mctrl(struct uart_port *port, unsigned int mctrl)
 	temp = lpuart32_read(port, UARTCTRL);
 	if (mctrl & TIOCM_LOOP)
 		temp |= UARTCTRL_LOOPS;
+	else
+		temp &= ~UARTCTRL_LOOPS;
+
 	lpuart32_write(port, temp, UARTCTRL);
 }
 
-- 
1.7.9.5

